src/Features/Core/Portable/ReplaceDocCommentTextWithTag/AbstractReplaceDocCommentTextWithTagCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,226 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.ReplaceDocCommentTextWithTag
+{
+    internal abstract class AbstractReplaceDocCommentTextWithTagCodeRefactoringProvider : CodeRefactoringProvider
+    {
+        protected abstract bool IsAnyKeyword(string text);
+        protected abstract bool IsXmlTextToken(SyntaxToken token);
+        protected abstract SyntaxNode ParseExpression(string text);
+
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var document = context.Document;
+            var span = context.Span;
+            var cancellationToken = context.CancellationToken;
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var token = root.FindToken(span.Start, findInsideTrivia: true);
+
+            if (!IsXmlTextToken(token))
+            {
+                return;
+            }
+
+            if (!token.FullSpan.Contains(span))
+            {
+                return;
+            }
+
+            var sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+
+            var singleWordSpan = ExpandSpan(sourceText, span, fullyQualifiedName: false);
+            var singleWordText = sourceText.ToString(singleWordSpan);
+            if (singleWordText == "")
+            {
+                return;
+            }
+
+            // First see if they're on a keyword. 
```

Comment:
```C#
    /// <summary>
    /// dynamic is a type
    /// </summary>
    class dynamic { }
```
Super bizarre case where you'd probably prefer the cref instead of the langword. I think we had a `class dynamic` at one point. 😉